vignettes/projpred.Rmd

```diff
@@ -0,0 +1,320 @@
+
+## Variable selection
+
+Now, **projpred** comes into play.
+```{r}
+library(projpred)
+```
+
+In **projpred**, the projective variable selection consists of a *search* part and an *evaluation* part. The search part determines the solution path, i.e., the best submodel for each model size^[Here, the model size is given by the number of predictor terms.]. The evaluation part determines the predictive performance of the submodels along the solution path. Depending on the search method, these two parts may also be entangled with each other.
+
+There are two functions for performing the variable selection: `varsel()` and `cv_varsel()`. In contrast to `varsel()`, `cv_varsel()` performs a cross-validation (CV) by running the search part with the training data of each CV fold separately (an exception is explained in the `?cv_varsel` help) and running the evaluation part on the corresponding test set of each CV fold. Because of this CV, `cv_varsel()` is recommended over `varsel()`. Thus, we use `cv_varsel()` here. Nonetheless, running `varsel()` first can offer a rough idea of the performance of the projections (i.e., of the submodels after projecting the reference model onto them). A more principled **projpred** workflow is work under progress.
+
+In versions > 2.0.2, **projpred** offers a parallelization of the projection. Typically, this only makes sense for a large number of projected draws. Therefore, this parallelization is not activated by a simple logical switch, but by a threshold for the number of projected draws below which no parallelization will be used. Values greater than or equal to this threshold will trigger the parallelization. For more information, see the general package documentation available at ``?`projpred-package` ``. There, we also explain why we are not running the parallelization on Windows and why we cannot recommend the parallelization of the projection for some types of reference models (see also section ["Supported types of reference models"](#refmodtypes) below).
+```{r}
+if (!identical(.Platform$OS.type, "windows")) {
+  trigger_default <- options(projpred.prll_prj_trigger = 200)
+  library(doParallel)
+  registerDoParallel(ncores)
+}
+```
+
+It is worth taking a look at the documentation of `cv_varsel()` because we cannot explain all arguments here. By modifying argument `cv_method`, we use a K-fold CV instead of a leave-one-out (LOO) CV. We do this here only to make this vignette build faster. In general, this is not recommended since LOO CV is more accurate. By modifying argument `nterms_max`, we impose a limit on the model size until which the search is continued. Typically, one has to run the variable selection with a large `nterms_max` first (the default value may not even be large enough) and only after inspecting the results from this first run, one is able to set a reasonable `nterms_max` in subsequent runs. The value we are using here (`12`) is based on such a first run (which is not shown here, though).
```

Comment:
"It is worth taking a look at the documentation of `cv_varsel()` because we cannot explain all arguments here."
->
Here, we use only some of the available arguments; see the documentation of `cv_varsel()` for the full list of arguments.

Reply:
"In general, this is not recommended since LOO CV is more accurate."
->
When the computation time is not an issue, we recommend using LOO-CV as it is likely to be more accurate.